### PR TITLE
Get Pylint Pass on Azure  Core

### LINKED ISSUE #5021: [azure-core]Need successful Pylint pass
The current azure-core code has a large volume of Pylint issues that need to be fixed.

### PR DESCRIPTION
**Pull request by rakshith91**
- Created new branch to rebase with latest changes and change of location

Resolves #5021 

Duplicate of #5152 with comments addressed.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/core/azure-core/azure/core/configuration.py [hunk 1]
```diff
@@ -52,6 +52,9 @@ def __init__(self, transport=None, **kwargs):
         # User Agent configuration
         self.user_agent_policy = None
 
+        # HTTP Transport
+        self.transport = transport
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/configuration.py]: Good catch

File: sdk/core/azure-core/azure/core/paging.py [hunk 2]
```diff
@@ -33,7 +33,6 @@
 from typing import Dict, Any, List, Callable, Optional, TYPE_CHECKING  # pylint: disable=unused-import
 
 if TYPE_CHECKING:
-    from .pipeline import HttpResponse  # pylint: disable=unused-import
     from msrest.serialization import Deserializer, Model  # pylint: disable=unused-import
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/paging.py]: It looks like a behavior change rather than a formatting change. Could you double check?

Comment (rakshith91) [reply sdk/core/azure-core/azure/core/paging.py]: good catch. i'll add this back. it's required for mypy anyway

**Pull request merged**